pkg/multiClusterWatch/watch.go

```diff
@@ -84,70 +94,147 @@ func (w *WatchController) WatchCluster(config *rest.Config) (Watcher, error) {
 	return watcher, nil
 }
 
-func (w *ClusterWatcher) Start(ctx context.Context) error {
-	log.Log.Info("Starting cluster watcher", "name", w.ClusterName)
-
-	informerFactory := informers.NewSharedInformerFactory(w.client, RESYNC_PERIOD)
+func (w *ClusterWatcher) WatchIngress(ctx context.Context, sharedInformer informers.SharedInformerFactory) {
 
-	informer := informerFactory.Networking().V1().Ingresses().Informer()
+	informer := sharedInformer.Networking().V1().Ingresses().Informer()
+	w.indexer = informer.GetIndexer()
 	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			log.Log.Info("got add event for ingress", "cluster watcher", w.ClusterName, "ingress", obj.(*networkingv1.Ingress).Namespace+"/"+obj.(*networkingv1.Ingress).Name)
-			current := obj.(*networkingv1.Ingress)
-			target := current.DeepCopy()
-			targetAccessor := traffic.NewIngress(target)
-			w.Handler.Handle(ctx, targetAccessor)
-			//todo handle requeue and errors
-			if !equality.Semantic.DeepEqual(current, target) {
-				//write back to cluster
-				w.client.NetworkingV1().Ingresses(target.Namespace).Update(ctx, target, metav1.UpdateOptions{})
-			}
+			w.Enqueue(obj)
 		},
 		UpdateFunc: func(old, obj interface{}) {
 			log.Log.Info("got update event for ingress", "cluster watcher", w.ClusterName, "ingress", obj.(*networkingv1.Ingress).Namespace+"/"+obj.(*networkingv1.Ingress).Name)
-			current := obj.(*networkingv1.Ingress)
-			target := current.DeepCopy()
-			targetAccessor := traffic.NewIngress(target)
-			w.Handler.Handle(ctx, targetAccessor)
-			//todo handle requeue and errors
-			if !equality.Semantic.DeepEqual(current, target) {
-				//write back to cluster
-				w.client.NetworkingV1().Ingresses(target.Namespace).Update(ctx, target, metav1.UpdateOptions{})
-			}
+			w.Enqueue(obj)
 		},
 		DeleteFunc: func(obj interface{}) {
 			log.Log.Info("got delete event for ingress", "cluster watcher", w.ClusterName, "ingress", obj.(*networkingv1.Ingress).Namespace+"/"+obj.(*networkingv1.Ingress).Name)
-			current := obj.(*networkingv1.Ingress)
-			target := current.DeepCopy()
-			targetAccessor := traffic.NewIngress(target)
-			w.Handler.Handle(ctx, targetAccessor)
-			//todo handle requeue and errors
-			if !equality.Semantic.DeepEqual(current, target) {
-				//write back to cluster
-				w.client.NetworkingV1().Ingresses(target.Namespace).Update(ctx, target, metav1.UpdateOptions{})
-			}
+			w.Enqueue(obj)
 		},
 	})
+}
 
+func (w *ClusterWatcher) Start(ctx context.Context) error {
+	defer runtimeUtil.HandleCrash()
+	defer w.Queue.ShutDown()
+	informerFactory := informers.NewSharedInformerFactory(w.client, RESYNC_PERIOD)
+
+	w.WatchIngress(ctx, informerFactory)
 	informerFactory.Start(ctx.Done())
 	informerFactory.WaitForCacheSync(ctx.Done())
 
 	log.Log.Info("started watcher events", "cluster watcher", w.ClusterName)
-
+	for i := 0; i < 1; i++ {
```

Comment:
not following the point of this for statement... it reads like it will only ever iterate the once, so why not remove it?

Reply:
sure. again  i think it is just from the original